### ISSUE TITLE
Automatic build.snapcraft.io rebuilds for outdated Ubuntu packages

### ISSUE THREAD

**Issue by OddBloke** [canonical-websites/build.snapcraft.io#1182]
This morning, I was scrubbing through my inbox and saw the semi-regular “SNAP_NAME contains outdated Ubuntu packages” notification. I’ve now manually kicked off a new build in build.snapcraft.io, which I will test and release once it’s complete.

It would be really cool if, at the same time the email was sent, build.snapcraft.io had automatically kicked off a new build of my snap; the build would have finished a couple of hours ago, and I would already be able to get on with the test/release phase, and my users would have the security-updated snap that much sooner.

Is this feasible?

(Apologies if this is the wrong place for filing bugs about more than just the front-end, but I'm getting no response on the forums so I thought this might be a better location.)

**Comment by evandandrea** [comment]
Yes, it's something we want to do but it hasn't quite bubbled to the top of the priority list yet. I thought we had an issue filed, but I couldn't find an existing one either.

**Comment by Lukewh** [comment]
This is now the issue to track this work, thanks :+1:!

**Comment by frederickjh** [comment]
Would it also be possible in the mean time to add a link to the email that would take one directly to the page with the button to manually rebuild the snap?  I find I have to click around too many times just to get there.

**Comment by cjwatson** [comment]
I'm not sure this is something I can fit in, sorry.  Unassigning myself.  If you need to get my time to work on this (though I'm not sure it's something I'm best-placed to do, anyway), then please ask Snap Store management.

It would probably be best if whatever sends the email notification also dispatched a webhook to BSI so that it can dispatch a build in turn, but I don't know what component sends the notification.

**Comment by bartaz** [comment]
@cjwatson Thanks. As far as I understand assigning you was not meant to mean we want you do to the works, but you seemed the person who may know the components that would need changing.

But from your answer I assume it's not Launchpad that sends the email, so maybe its the Store itself.

**Comment by cjwatson** [comment]
@bartaz OK - @shivjoshi1996, please explain this sort of thing since otherwise a surprise assignment comes as, er, a surprise.

Indeed, Launchpad doesn't send this email.  It's also not the main store code AFAICS, although somebody on the store team should know.

**Comment by shivjoshi1996** [comment]
Apologies for the confusion @cjwatson, we were quickly trying to triage some bugs and I assigned you, expecting someone else to comment and provide you with some context. Next time we will just comment on the issue with what we need to ask you. Cheers.

**Comment by frederickjh** [comment]
When I commented about adding a link in the email to where one could to rebuild the affected snap would need to look like this to the user account listing snaps and status:

[https://build.snapcraft.io/user/username](https://build.snapcraft.io/user/username)

or directly to the snap project

[https://build.snapcraft.io/user/username/snapname](https://build.snapcraft.io/user/username/snapname)


Currently the only link in the email comes at the end and is a link to the security release information and looks something like this:
[https://usn.ubuntu.com/3867-1/](https://usn.ubuntu.com/3867-1/)

**Comment by bartaz** [comment]
We may need to investigate, but it seems that part of snap ecosystem that detects outdated Ubuntu dependencies and sends this email is completely independent (and unaware) of build.snapcraft.io. This email quite likely is sent regardless of the way snap is built (via build.snapcraft or manually).
